h2/src/test/org/h2/test/db/TestSessionsLocks.java

```diff
@@ -142,4 +145,57 @@ private void testCancelStatement() throws Exception {
         conn.close();
     }
 
+    private void testAbortStatement() throws Exception {
+        deleteDb("sessionsLocks");
+        Connection conn = getConnection("sessionsLocks");
+        Statement stat = conn.createStatement();
+        ResultSet rs;
+        rs = stat.executeQuery("select session_id() as ID");
+        rs.next();
+        int sessionId = rs.getInt("ID");
+
+        // Setup session to be aborted
+        Connection conn2 = getConnection("sessionsLocks");
+        Statement stat2 = conn2.createStatement();
+        stat2.execute("create table test(id int primary key, name varchar)");
+        conn2.setAutoCommit(false);
+        stat2.execute("insert into test values(1, 'Hello')");
+        conn2.commit();
+        // grab a lock
+        stat2.executeUpdate("update test set name = 'Again' where id = 1");
+
+        rs = stat2.executeQuery("select session_id() as ID");
+        rs.next();
+
+        int otherId = rs.getInt("ID");
+        assertTrue(otherId != sessionId);
+        assertFalse(rs.next());
+
+        // expect one lock
+        assertEquals(1, getLockCountForSession(stat, otherId));
+        rs = stat.executeQuery("CALL ABORT_SESSION(" + otherId + ")");
+        rs.next();
+        assertTrue(rs.getBoolean(1));
+
+        // expect the lock to be released along with its session
+        assertEquals(0, getLockCountForSession(stat, otherId));
+        rs = stat.executeQuery("CALL ABORT_SESSION(" + otherId + ")");
+        rs.next();
+        assertFalse("Session is expected to be already aborted", rs.getBoolean(1));
+
+        // using the connection for the aborted session is expected to throw an
+        // exception
+        assertThrows(ErrorCode.DATABASE_CALLED_AT_SHUTDOWN, stat2).executeQuery("select count(*) from test");
```

Comment:
It looks like you need to replace `ErrorCode.DATABASE_CALLED_AT_SHUTDOWN` with `config.networked ? ErrorCode.CONNECTION_BROKEN_1 : ErrorCode.DATABASE_CALLED_AT_SHUTDOWN`.

Reply:
I don't think its so simple, as this the 4th run of the test. I will dig a bit deeper, likely because of timing we possibly cannot rely on a specific exception to be thrown.

Reply:
Each iteration uses different settings, they aren't the same. The failure was detected in networked test only and because you close the session the `CONNECTION_BROKEN_1` looks like the most expected exception on the client side.

You can change the related flag in `TestAll` and run only this test to be sure.

Reply:
Thanks for your suggestion, this resolves. Root cause is that SessionRemote.isClosed() behaves different from the Session.isClosed().